src/main/java/de/tum/in/www1/artemis/domain/notification/GroupNotificationFactory.java

```diff
@@ -165,59 +199,100 @@
+            NotificationType notificationType, Course course) {
+        String title;
+        String text;
+        String[] placeholderValues;
+        GroupNotification notification;
+
+        switch (notificationType) {
             case NEW_REPLY_FOR_EXERCISE_POST -> {
                 Exercise exercise = post.getExercise();
                 title = NEW_REPLY_FOR_EXERCISE_POST_TITLE;
-                text = "Exercise \"" + exercise.getTitle() + "\" got a new reply.";
-                notification = new GroupNotification(course, title, text, author, groupNotificationType);
+                text = NEW_REPLY_FOR_EXERCISE_POST_TEXT;
+                placeholderValues = ArrayUtils.addAll(generatePlaceholderValuesForMessageNotificationsWithAnswers(course, post, answerPost), exercise.getTitle());
+                notification = new GroupNotification(course, title, text, true, placeholderValues, author, groupNotificationType);
                 notification.setTransientAndStringTarget(createExercisePostTarget(post, course));
             }
             case NEW_REPLY_FOR_LECTURE_POST -> {
                 Lecture lecture = post.getLecture();
                 title = NEW_REPLY_FOR_LECTURE_POST_TITLE;
-                text = "Lecture \"" + lecture.getTitle() + "\" got a new reply.";
-                notification = new GroupNotification(course, title, text, author, groupNotificationType);
+                text = NEW_REPLY_FOR_LECTURE_POST_TEXT;
+                placeholderValues = ArrayUtils.addAll(generatePlaceholderValuesForMessageNotificationsWithAnswers(course, post, answerPost), lecture.getTitle());
+                notification = new GroupNotification(course, title, text, true, placeholderValues, author, groupNotificationType);
                 notification.setTransientAndStringTarget(createLecturePostTarget(post, course));
             }
             case NEW_REPLY_FOR_COURSE_POST -> {
                 title = NEW_REPLY_FOR_COURSE_POST_TITLE;
-                text = "Course-wide post in course \"" + course.getTitle() + "\" got a new reply.";
-                notification = new GroupNotification(course, title, text, author, groupNotificationType);
+                text = NEW_REPLY_FOR_COURSE_POST_TEXT;
+                placeholderValues = ArrayUtils.addAll(generatePlaceholderValuesForMessageNotificationsWithAnswers(course, post, answerPost));
+                notification = new GroupNotification(course, title, text, true, placeholderValues, author, groupNotificationType);
                 notification.setTransientAndStringTarget(createCoursePostTarget(post, course));
             }
             default -> throw new UnsupportedOperationException("Unsupported NotificationType: " + notificationType);
         }
         return notification;
     }
 
+    private static String[] generatePlaceholderValuesForMessageNotifications(Course course, Post post) {
```

Comment:
This code has a 100% duplicate in SingleUserNotification.

Reply:
I refactored it.